src/services/dashboards/dashboard-detail/modules/DashboardWidgetContainer.vue

```diff
@@ -25,50 +30,84 @@
 <script lang="ts">
 import type { PropType, SetupContext } from 'vue';
 import {
-    defineComponent, reactive, toRefs, ref, onMounted, watch, onBeforeUnmount,
+    defineComponent, reactive, toRefs, ref, onMounted, watch, onBeforeUnmount, computed,
 } from 'vue';
 
+import { flattenDeep } from 'lodash';
+
 import {
-    WIDGET_CONTAINER_MAX_WIDTH,
-    WIDGET_CONTAINER_MIN_WIDTH,
+    WIDGET_CONTAINER_MAX_WIDTH, WIDGET_CONTAINER_MIN_WIDTH,
 } from '@/services/dashboards/dashboard-detail/lib/config';
 import { widgetThemeAssigner } from '@/services/dashboards/dashboard-detail/lib/theme-helper';
-import type { WidgetThemeAssignedList, WidgetThemeOption } from '@/services/dashboards/dashboard-detail/lib/type';
 import { widgetWidthAssigner } from '@/services/dashboards/dashboard-detail/lib/width-helper';
 import AWSCloudFrontCost from '@/services/dashboards/widgets/aws-cloud-front-cost/AWSCloudFrontCost.vue';
-import type { WidgetSize } from '@/services/dashboards/widgets/config';
+import type { WidgetSize, DashboardLayoutWidgetInfo, WidgetConfig } from '@/services/dashboards/widgets/config';
 import { WIDGET_SIZE } from '@/services/dashboards/widgets/config';
+import type { WidgetTheme } from '@/services/dashboards/widgets/view-config';
+import { getWidgetConfig } from '@/services/dashboards/widgets/widget-helper';
 
 
-export default defineComponent({
+interface ContainerWidgetConfig extends WidgetConfig {
+    size: WidgetSize;
+    width: number;
+    themeName?: WidgetTheme;
+}
+interface Props {
+    dashboardWidgetLayouts: DashboardLayoutWidgetInfo[][];
+}
+export default defineComponent<Props>({
     name: 'DashboardWidgetContainer',
     components: {
         AWSCloudFrontCost,
     },
     props: {
-        widgetSizeList: {
-            type: Array as PropType<Array<WidgetSize>>,
-            default: () => ([]),
-        },
-        widgetThemeOptionList: {
-            type: Array as PropType<Array<WidgetThemeOption>>,
+        dashboardWidgetLayouts: {
+            type: Array as PropType<DashboardLayoutWidgetInfo[][]>,
             default: () => ([]),
         },
     },
     setup(props, { emit, expose }: SetupContext) {
         const state = reactive({
             // width
             containerWidth: WIDGET_CONTAINER_MIN_WIDTH,
-            widgetSizeList: props.widgetSizeList,
-            widgetWidthList: [] as Array<Array<number>>,
-            widgetIndexList: [] as Array<Array<number>>,
+            widgetSizeList: [] as WidgetSize[],
```

Comment:
description:
I followed **single source of truth**.
Only change `widgetSizeList ` and let the other states be computed accordingly.